src/IcalParser.php

```diff
@@ -626,7 +629,7 @@ public function getEvents($includeRecurring=false) {
 						foreach($recurrences as $j => $recurDate) {
 							$newEvent = $event;
 							// DateTime inequality must be determined with != instead of !==
-							if ($recurDate != $event['DTSTART']) {
+							if ($newEvent && ($recurDate != $event['DTSTART'])) {
```

Comment:
Again, just checking for a falsy value which might occur due to which events exist in the filtered list of current events passed to this method.